### ISSUE TITLE
neural: Hebbian edges evaporate (~12 min half-life) — decay_rate applied per-second; compounding never accumulates

### ISSUE THREAD

**Issue by JFK** [kagura-ai/memory-cloud#970]
## Symptom

Hebbian edges never accumulate. On production (measured 2026-06-10, active=green):

- `neural_memory_edges` by origin: **semantic 4,433 (91%)**, declared 375 (8%), **hebbian 61 → 32 → 38** (churning within one hour).
- **Every** hebbian edge is < 33 min old (`min(created_at)` = 33 min before query time); **none survive from prior days**.
- 32-min-old edges already sit at weight **0.027–0.043**, heading for the prune threshold.
- The most-used context (`kagura-dev`, 2,292 memories) has **0** hebbian edges; a small, focused context (`kagura-engineer-dev`, 88 memories) accrued 24 during a single session.

Net effect: the "cells that fire together, wire together" layer is a **short-lived echo of the current session**, not a durable, compounding structure. This directly blocks the compounding-data moat (and #969 Tier B, which cannot demonstrate cold→warm lift while edges evaporate).

## Root cause — decay is applied per-second with a per-(longer-unit) rate

`backend/src/neural/decay.py:72`:

```python
decay_factor = math.exp(-self.config.decay_rate * delta_seconds)
```

- `decay_rate = 0.001` (config.py:140), `delta_seconds` is **raw elapsed seconds** (≈ `decay_background_interval = 3600` per run).
- Each background pass multiplies every Hebbian weight by `exp(-0.001 × 3600) = exp(-3.6) = 0.0273` — a **97.3% loss per hour**.
- **Half-life = ln(2) / 0.001 = 693 s ≈ 11.6 minutes.** Even a maxed edge (w=0.4) crosses the prune threshold (0.01 code default / 0.001 in prod DB) within ~1–2 hours without re-reinforcement.
- Since `learning_rate = 0.05` produces Δw ≈ 0.05 per co-activation, an edge would need to be re-co-activated several times **within minutes** just to stay alive — impossible under normal "ask once, move on" usage.

The intended unit is almost certainly **not seconds**. The sibling temporal config makes this explicit:

```python
recency_tau_days: float = 14.0  # 2-week half-life   (config.py:126)
```

Recency decay is expressed in **days**; the neural design clearly intends slow, multi-week forgetting. With `decay_rate = 0.001`, a sensible half-life only appears if Δt is in **hours** (half-life ≈ 28.9 days) or the rate is recalibrated.

Issue #722 made semantic/declared edges **decay-exempt**, so only Hebbian edges evaporate — which is exactly why the graph is now 91% similarity-derived (usage-independent) and ~1% usage-derived. The moat mechanism is the only thing being deleted.

## Secondary bug — the elapsed-time logic is dead

`tasks/neural_tasks.py` constructs a **fresh `DecayManager` per task run** (inside the per-graph loop), so `self._last_decay_time` is always `None`. Every run therefore falls into the first-run branch and uses `delta_seconds = decay_background_interval` (3600), regardless of how much time actually elapsed. `graph_model.last_decay_at` is written to the DB but never read back into the manager. If the task schedule were changed (e.g. every 5 min), each run would still apply a full 3600 s of decay → even more aggressive. The "time delta" computation in `apply_decay` is effectively inert.

## Impact

- **Moat-relevant:** the compounding-data moat ("the longer you use it, the better it remembers") cannot materialize — the learned layer is wiped hourly. This is the root cause behind the corpus finding that drove #967/#969.
- **Blocks #969** (Tier B compounding proof): replaying co-recall traffic will not produce lasting edges until this is fixed.
- No impact on #967 Tier A (semantic + Sleep are decay-exempt and intact).

## Suggested fix

Prefer making the unit explicit and consistent with `recency_tau_days`:

- **(a) Half-life config (recommended):** add `hebbian_decay_half_life_days` (default ~14–30, matching the recency intuition) and derive `decay_factor = 0.5 ** (delta_seconds / (half_life_days * 86400))`. Self-documenting, no magic constants.
- **(b) Convert delta to days:** `delta_days = delta_seconds / 86400` and set `decay_rate` to ~0.05/day (≈14-day half-life).
- **(c) Keep seconds, recalibrate default:** `decay_rate ≈ 2.7e-7` for a ~30-day half-life (opaque; least preferred).

Also fix the secondary bug: persist/read `last_decay_at` so `delta_seconds` reflects real elapsed time, or remove the dead time-delta logic and decay by the known fixed interval deliberately.

## Validation

After the fix, confirm on a warmed context that Hebbian edges (i) persist across days and (ii) grow with repeated co-recall (weight trending up, count > 2), and that `min(created_at)` for hebbian edges predates the current session. Add a regression test asserting half-life ≈ configured value (e.g. weight after `half_life` seconds ≈ 0.5·w₀).

---
_From the 2026-06-10 moat strategy review (corpus audit → hebbian evaporation root-cause). Recorded in kagura-saas-strategy / kagura-ecosystem memory. Blocker for #969._
